src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.netcoreapp.cs

```diff
@@ -91,67 +92,107 @@ public void Dictionary_Generic_RemoveKey_DefaultKeyContainedInDictionary(int cou
+            Assert.Equal(capacity, dictionary.EnsureCapacity(capacity));
+
+            dictionary = new Dictionary<int, int>();
+            for (int i = 0; i < capacity; i++)
+            {
+                dictionary.Add(i, 0);
+            }
+            Assert.Equal(capacity, dictionary.EnsureCapacity(capacity));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(7)]
+        public void EnsureCapacity_EnsureCapacityCalledTwice_ReturnsSameValue(int count)
+        {
+            var dictionary = new Dictionary<int, int>();
+            for (int i = 0; i < count; i++)
+            {
+                dictionary.Add(i, 0);
+            }
+            int capacity = dictionary.EnsureCapacity(count * 2);
+            Assert.Equal(capacity, dictionary.EnsureCapacity(count * 2));
         }
         
         [Theory]
         [InlineData(1)]
         [InlineData(5)]
-        public void EnsureCapacity_CapacityRequestedSmallerThanCount_SetsCapacityToLargerOrEqualToExistingCount(int count)
+        [InlineData(7)]
+        public void EnsureCapacity_DictionaryNotEmpty_RequestedSmallerThanCount_ReturnsAtLeastSizeOfCount(int count)
         {
             var dictionary = new Dictionary<int, int>();
             for (int i = 0; i < count; i++)
             {
-                dictionary.Add(i, i);
+                dictionary.Add(i, 0);
             }
-            Assert.InRange(dictionary.EnsureCapacity(count - 1), dictionary.Count, int.MaxValue);
+            Assert.InRange(dictionary.EnsureCapacity(count - 1), count, int.MaxValue);
         }
 
         [Fact]
-        public void EnsureCapacity_GivenNonEmptyDictionary_CapacityRemainsTheSameUnlessCalledWithValueLargerThanExisting()
+        public void EnsureCapacity_DictionaryNotEmpty_SetsToAtLeastTheRequested()
         {
             int count = 20;
             var dictionary = new Dictionary<int, int>();
             for (int i = 0; i < count; i++)
             {
-                dictionary.Add(i, i);
+                dictionary.Add(i, 0);
             }
-
-            // assert capacity won't change when ensuring a capacity smaller or equal than existing
+            
+            // get current capacity
             int currentCapacity = dictionary.EnsureCapacity(0);
-            Assert.Equal(currentCapacity, dictionary.EnsureCapacity(currentCapacity));
-            Assert.Equal(currentCapacity, dictionary.EnsureCapacity(currentCapacity - 2));
 
             // assert we can update to a larger capacity
             int newCapacity = dictionary.EnsureCapacity(currentCapacity * 2);
             Assert.InRange(newCapacity, currentCapacity * 2, int.MaxValue);
-
-            // assert new capacity remains the newly updated one
-            Assert.Equal(newCapacity, dictionary.EnsureCapacity(0));
         }
 
         [Fact]
-        public void EnsureCapacity_SetsToNextPrimeNumber()
+        public void EnsureCapacity_CapacityIsSetToPrimeNumberLargerOrEqualToRequested()
```

Comment:
Improving the working on this test case since we don't always set to next prime. (we skip 5, 13, etc.)